smack-extensions/src/test/java/org/jivesoftware/smackx/muc/MultiUserChatTest.java

```diff
@@ -0,0 +1,211 @@
+                assertNotNull(groupChatInvitation, "Missing GroupChatInvitation extension");
+                assertEquals(roomJid, groupChatInvitation.getRoomAddress());
+
+                updateRequestSent.set(true);
+            }
+        };
+        serverSimulator.start();
+
+        // Create multi user chat
+        MultiUserChat multiUserChat = multiUserChatManager.getMultiUserChat(roomJid);
+
+        // Call tested method
+        multiUserChat.inviteDirectly(userJid);
+
+        // Wait for processing requests
+        serverSimulator.join(RESPONSE_TIMEOUT_IN_MILLIS);
+        serverSimulator.complete();
+
+        // Check if an error occurred within the simulator
+        final Throwable exception = serverSimulator.getException();
+        if (exception != null) {
+            throw exception;
+        }
+
+        assertTrue(updateRequestSent.get(), "Invite directly request not sent");
+    }
+
+    @Test
+    public void shouldReceiveOfflineInvitation() throws XmppStringprepException {
+        EntityBareJid roomJid = JidCreate.entityBareFrom("room@example.com");
+        EntityFullJid inviterJid = JidCreate.entityFullFrom("inviter@example.com/user1");
+        EntityBareJid inviteeJid = JidCreate.entityBareFrom("invitee@example.com");
+
+        GroupChatInvitation groupChatInvitation = new GroupChatInvitation(JidCreate.entityBareFrom("room@example.com"));
+        Message sentMessage = MessageBuilder.buildMessage(UUID.randomUUID().toString())
+                .from(inviterJid)
+                .to(inviteeJid)
+                .addExtension(groupChatInvitation)
+                .build();
+
+        // Prepare listener to receive a group invitation
+        GroupInvitationListener groupInvitationListener = new GroupInvitationListener() {
+            @Override
+            public void verifyInvitation(XMPPConnection conn, MultiUserChat room, EntityJid inviter,
+                                         String reason, String password, Message message, MUCUser.Invite invitation) {
+                // Check all parameters values.
+                assertSame(connection, conn);
+                assertSame(connection, room.getXmppConnection());
+                assertEquals(roomJid, room.getRoom());
+                assertEquals(inviterJid, inviter);
+                assertNull(reason);
+                assertNull(password);
+                assertSame(sentMessage, message);
+                assertNull(invitation);
+            }
+        };
+        multiUserChatManager.addInvitationListener(groupInvitationListener);
+
+        // Simulate sending a message with a group invitation
+        connection.processStanza(sentMessage);
+
+        // Wait for the listener to be called or throw a timeout exception
+        groupInvitationListener.waitUntilInvocationOrTimeout();
+    }
+
+    private static class Responder extends Thread {
+        protected final AtomicBoolean running = new AtomicBoolean();
```

Comment:
This field isn't used anywhere, in fact the whole `Responder` class could be merged into `InvokeDirectlyResponder`. I assume this was potentially at some point in the past different, but right now it is not required in this form and hence only contributes to confusion, i.e., it makes the code harder to read.